library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsMediaPeriod.java

```diff
@@ -454,20 +454,25 @@ private void buildAndPrepareSampleStreamWrappers(long positionUs) {
             : Collections.emptyMap();
 
     boolean hasVariants = !masterPlaylist.variants.isEmpty();
+    boolean hasIFrameVariants = !masterPlaylist.iFrameVariants.isEmpty();
     List<Rendition> audioRenditions = masterPlaylist.audios;
     List<Rendition> subtitleRenditions = masterPlaylist.subtitles;
 
     pendingPrepareCount = 0;
     ArrayList<HlsSampleStreamWrapper> sampleStreamWrappers = new ArrayList<>();
     ArrayList<int[]> manifestUrlIndicesPerWrapper = new ArrayList<>();
 
-    if (hasVariants) {
+    ArrayList<Variant> allVariants = new ArrayList<>();
+    allVariants.addAll(masterPlaylist.variants);
+    allVariants.addAll(masterPlaylist.iFrameVariants);
+    if (hasVariants){
       buildAndPrepareMainSampleStreamWrapper(
           masterPlaylist,
           positionUs,
           sampleStreamWrappers,
           manifestUrlIndicesPerWrapper,
-          overridingDrmInitData);
+          overridingDrmInitData,
+          allVariants);
     }
```

Comment:
Sorry the diff might be a bit hard to read.  Basically added an argument to pass Variants and replaced all references to the `variants` field in the `HlsMasterPlaylist` with same.